### PR TITLE
Add ability to compare visual state of form with pre-saved dump

### PR DESCRIPTION
**Pull request by mialeska**
- created form interface and implementation
- implemented DumpManager to save the dump and compare with it
- moved VisualStateProvider to Visualization namespace
- moved LogElementState delegates to Logging namespace

### CODE REVIEW

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Forms/Form.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    public abstract class Form<T> : IForm where T : IElement
+    {
+        /// <summary>
+        /// Name of the current form.
+        /// </summary>
+        public abstract string Name { get; }
+
+        /// <summary>
+        /// Visualization configuration used by <see cref="Dump"/>.
+        /// Could be get from AqualityServices.
```
Comment (n-verbitsky) [inline Aquality.Selenium.Core/src/Aquality.Selenium.Core/Forms/Form.cs:26]: A slight typo - should be `could be gotten`. However, I would prefer `Could be obtained`.

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Configurations/VisualizationConfiguration.cs [hunk 2]
```diff
@@ -24,5 +25,14 @@ public VisualizationConfiguration(ISettingsFile settingsFile)
         public int ComparisonWidth => settingsFile.GetValueOrDefault(".visualization.comparisonWidth", 16);
 
         public int ComparisonHeight => settingsFile.GetValueOrDefault(".visualization.comparisonHeight", 16);
+
+        public string PathToDumps
+        {
+            get
+            {
+                var pathInConfiguration = settingsFile.GetValueOrDefault(".visualization.pathToDumps", "../../../Resources/Dumps/");
```
Comment (n-verbitsky) [inline Aquality.Selenium.Core/src/Aquality.Selenium.Core/Configurations/VisualizationConfiguration.cs:33]: I guess, `Resources/Dumps` might be not clear enough to indicate its purpose. @mialeska, how about `Resources/VisualTesting/Dumps` or `Resources/VisualDumps`?

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Forms/Form.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        /// </summary>
+        protected abstract IVisualizationConfiguration VisualizationConfiguration { get; }
+
+        /// <summary>
+        /// Localizer logger used by <see cref="Dump"/>.
```
Comment (n-verbitsky) [inline Aquality.Selenium.Core/src/Aquality.Selenium.Core/Forms/Form.cs:31]: Small typo - should be `Localized logger`

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Forms/Form.cs (hunk 1 above)
Comment (n-verbitsky) [reply Aquality.Selenium.Core/src/Aquality.Selenium.Core/Forms/Form.cs:26]: Seemingly, needs to be updated further as well.

**Review by n-verbitsky — APPROVED** [review]

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Forms/Form.cs (hunk 3 above)
Comment (mialeska) [reply Aquality.Selenium.Core/src/Aquality.Selenium.Core/Forms/Form.cs]: fixed

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Configurations/VisualizationConfiguration.cs (hunk 2 above)
Comment (mialeska) [reply Aquality.Selenium.Core/src/Aquality.Selenium.Core/Configurations/VisualizationConfiguration.cs]: Resources/VisualDumps works for me

File: Aquality.Selenium.Core/src/Aquality.Selenium.Core/Forms/Form.cs (hunk 1 above)
Comment (mialeska) [reply Aquality.Selenium.Core/src/Aquality.Selenium.Core/Forms/Form.cs]: updated to "Could be obtained"

**Pull request merged**